course/src/isometry.cc

```diff
@@ -0,0 +1,115 @@
+#include <isometry.h>
+
+namespace ekumen {
+  namespace math {
+
+  const Vector3 Vector3::kUnitX(1.,0,0);
+  const Vector3 Vector3::kUnitY(0,1.,0);
+  const Vector3 Vector3::kUnitZ(0,0,1.);
+  const Vector3 Vector3::kZero(0,0,0);
+
+  Vector3 Vector3::operator + (const Vector3 & obj) const
+  {
+    return Vector3( x_ + obj.x(), y_ + obj.y(),z_ +  obj.z());
+  }
+
+  Vector3 Vector3::operator - (const Vector3 & obj) const
+  {
+    return Vector3( x_ - obj.x(), y_ - obj.y(),z_ -  obj.z());
+  }
+
+  bool Vector3::operator == (const std::initializer_list<double> & obj) const
+  {
+    if (obj.size() != 3) return false;
+    int i = 0;
+    for (auto & e : obj)
+    {
+      if (*data_[i] != e) return false;
```

Comment:
This is "legal" syntax, but prone to programmer errors. Always use `{ }` around a code scope, and don't use same-line fragments like this. It's hard to spot when you're trying to understand visually code someone else wrote from a multi-thousand lines files.

Structure your code so that blocks can be _visually identified_ before even reading the details. It's great helper when you're browsing a large file.